bot/seasons/evergreen/minesweeper.py

```diff
@@ -0,0 +1,212 @@
+            msg = await ctx.send(f"{ctx.author.mention} you already have a game running")
+            await msg.delete(delay=2)
+            await ctx.message.delete(delay=2)
+            return
+
+        # Add game to list
+        board: GameBoard = self.generate_board(bomb_chance)
+        revealed_board: GameBoard = [["hidden" for _ in range(10)] for _ in range(10)]
+
+        await ctx.send(f"{ctx.author.mention} is playing minesweeper")
+        chat_msg = await ctx.send(self.format_for_discord(revealed_board))
+
+        await ctx.author.send("play by typing: `.reveal xy [xy]` or `.flag xy [xy]` \n"
+                              "close the game with `.end`\n"
+                              "cords must be in format `<letter><number>`")
+        dm_msg = await ctx.author.send(self.format_for_discord(revealed_board))
+
+        self.games[ctx.author.id] = {
+            "board": board,
+            "revealed": revealed_board,
+            "dm_msg": dm_msg,
+            "chat_msg": chat_msg
+        }
+
+    async def reload_board(self, ctx: commands.Context) -> None:
+        """Update both playing boards."""
+        game = self.games[ctx.author.id]
+        await game["dm_msg"].delete()
+        game["dm_msg"] = await ctx.author.send(self.format_for_discord(game["revealed"]))
+        await game["chat_msg"].edit(content=self.format_for_discord(game["revealed"]))
+
+    @commands.dm_only()
+    @commands.command(name="flag")
+    async def flag_command(self, ctx: commands.Context, *cords: CordConverter) -> None:
+        """Place multiple flags on the board"""
+        board: GameBoard = self.games[ctx.author.id]["revealed"]
+        for x, y in cords:
+            if board[y][x] == "hidden":
+                board[y][x] = "flag"
+
+        await self.reload_board(ctx)
+
+    async def lost(self, ctx: commands.Context) -> None:
+        """The player lost the game"""
+        game = self.games[ctx.author.id]
+        game["revealed"] = game["board"]
+        await self.reload_board(ctx)
+        await ctx.author.send(":fire: You lost :fire: ")
+        await game["chat_msg"].channel.send(f":fire: {ctx.author.mention} just lost minesweeper :fire:")
+        del self.games[ctx.author.id]
+
+    async def won(self, ctx: commands.Context) -> None:
+        """The player won the game"""
+        game = self.games[ctx.author.id]
+        game["revealed"] = game["board"]
+        await self.reload_board(ctx)
+        await ctx.author.send(":tada:  You won! :tada: ")
+        await game["chat_msg"].channel.send(f":tada: {ctx.author.mention} just won minesweeper :tada:")
+        del self.games[ctx.author.id]
+
+    def reveal_zeros(self, revealed: GameBoard, board: GameBoard, x: int, y: int) -> None:
+        """Used when a 0 is encountered to do a flood fill"""
+        for x_ in [x - 1, x, x + 1]:
+            for y_ in [y - 1, y, y + 1]:
+                if x_ == -1 or x_ == 10 or y_ == -1 or y_ == 10 or revealed[y_][x_] != "hidden":
```

Comment:
This is similar to the code in `generate_board`. I think it'd be good to put this in a generic search function which yields the matching coordinates.